types/jsreport-core/index.d.ts

```diff
@@ -2,95 +2,255 @@
+		width: string;
+		height: string;
+		headerHeight: string;
+		footerHeight: string;
+		format: string;
+		orientation: 'portrait' | 'landscape';
+		blockJavaScript: boolean;
+		resourceTimeout: number;
+		waitForJS: boolean;
+		fitToPage: boolean;
+		customPhantomJS: boolean;
+		phantomjsVersion: string;
 	}
 
-	interface JsReporter {
-		afterRenderListeners: Listener;
-		afterTemplatingEnginesExecutedListeners: Listener;
-		beforeRenderListeners: Listener;
-		documentStore: DocumentStore;
-		initializeListeners: Listener;
-		logger: Logger;
-		validateRenderListeners: Listener;
-		init(): Promise<void>;
-		render(options: RenderOptions): Promise<Report>;
-		use(extension: any): any;
+	interface Template {
+		phantom?: Partial<Phantom>;
+	}
+}
+
+declare namespace JsReportPhantomPdf {
+	const enum PhantomStrategy {
+		dedicatedProcess = 'dedicated-process',
+		phantomServer = 'phantom-server'
+	}
+
+	interface Options {
+		allowLocalFilesAccess: boolean;
+		// appDirectory: string;
+		defaultPhantomjsVersion: string;
+		strategy: PhantomStrategy;
+		timeout: number;
+	}
+
+	// without exporting enum, it doesn't include the require('jsreport-core') in the test.js for some reason
+	// help welcome
+	export enum Foo { }
+}
+
+declare function JsReportPhantomPdf(options?: Partial<JsReportPhantomPdf.Options>): JsReport.ExtensionDefinition;
+
+declare module 'jsreport-phantom-pdf' {
+	export = JsReportPhantomPdf;
+}
+
+declare namespace JsReport {
+	const enum Recipe {
+		Xlsx = 'xlsx'
+	}
+
+	interface Xlsx {
+		shortid: string;
 	}
+
+	interface Template {
+		xlsxTemplate: Partial<Xlsx>;
+	}
+}
+
+declare namespace JsReportXlsx {
+	interface Options {
+		addBufferSize: number;
+		escapeAmp: boolean;
+		numberOfParsedAddIterations: number;
+	}
+
+	// without exporting enum, it doesn't include the require('jsreport-core') in the test.js for some reason
+	// help welcome
+	export enum Foo { }
 }
 
-declare function JsReport(options?: Partial<{
-	autoTempCleanup: boolean;
-	dataDirectory: string;
-	extensionsLocationCache: boolean;
-	loadConfig: boolean;
-	logger: {
-		silent: boolean;
-	};
-	rootDirectory: string;
-	scripts: {
-		allowedModules: string[];
-	};
-	tasks: {
-		[task: string]: any;
-	};
-	tempDirectory: string;
-}>): JsReport.JsReporter;
-
-export = JsReport;
+declare function JsReportXlsx(options?: Partial<JsReportXlsx.Options>): JsReport.ExtensionDefinition;
+
+declare module 'jsreport-xlsx' {
+	export = JsReportXlsx;
+}
+
+declare namespace JsReport {
+	const enum Recipe {
+		Html2Xlsx = 'html-to-xlsx'
+	}
+}
+
+declare namespace JsReportHtml2Xlsx {
+	interface Options extends JsReportXlsx.Options {
+		strategy: string;
+	}
+
+	// without exporting enum, it doesn't include the require('jsreport-core') in the test.js for some reason
+	// help welcome
+	export enum Foo { }
+}
+
+declare function JsReportHtml2Xlsx(options?: Partial<JsReportHtml2Xlsx.Options>): JsReport.ExtensionDefinition;
+
+declare module 'jsreport-html-to-xlsx' {
```

Comment:
Instead of modifying this declaration file to no longer be a module, it would be more appropriate to create respective packages